libethereum/ChainParams.cpp

```diff
@@ -135,48 +97,33 @@ ChainParams ChainParams::loadConfig(
 	cp.allowFutureBlocks = params.count(c_allowFutureBlocks);
 
 	// genesis
-	string genesisStr = json_spirit::write_string(obj[c_genesis], false);
-	cp = cp.loadGenesis(genesisStr, _stateRoot);
+    string genesisStr = js::write_string(obj[c_genesis], false);
+    cp = cp.loadGenesis(genesisStr, _stateRoot);
 	// genesis state
-	string genesisStateStr = json_spirit::write_string(obj[c_accounts], false);
+    string genesisStateStr = js::write_string(obj[c_accounts], false);
 
     cp.genesisState = jsonToAccountMap(
         genesisStateStr, cp.accountStartNonce, nullptr, &cp.precompiled, _configPath);
-    cp.stateRoot = _stateRoot ? _stateRoot : cp.calculateStateRoot(true);
 
-    return cp;
-}
+    // Strict account check
+    js::read_string_or_throw(genesisStateStr, val);
+    for (auto const& account: val.get_obj())
+        validateAccountObj(account.second.get_obj());
```

Comment:
Do you need his here at all? You call `validateConfigJson(obj);` above and inside it it already goes over all accounts, no?

Reply:
you right